src/compiler/checker.ts

```diff
@@ -4504,18 +4526,34 @@ namespace ts {
                         parameters.push(paramSymbol);
                     }
 
-                    if (param.type && param.type.kind === SyntaxKind.StringLiteralType) {
-                        hasStringLiterals = true;
-                    }
+                    if (isJSDocParameterTag(param)) {
+                        if (param.typeExpression && param.typeExpression.type && param.typeExpression.type.kind === SyntaxKind.StringLiteralType) {
+                            hasStringLiterals = true;

```

Comment:
Weirdly enough, this doesn't really affect anything in the type system, since specialized signatures only rejigger the order of overloads, and I don't think these JSDoc tags really cover overloads.

You could consider removing this code, but I'd add a test if you're keeping it in.